<?xml version="1.0"?>
<launch>

    <!-- Parametrized sample launch file to run pybullet ros wrapper -->

    <!-- partial configuration params for pybullet_ros node, rest will be loaded from config_file -->
    <arg name="config_file" default="$(find pybullet_ros)/ros/config/pybullet_params_example.yaml"/>
    <arg name="plugin_import_prefix" default="pybullet_ros.plugins"/>
    <arg name="environment" default="environment"/>
    <arg name="pybullet_gui" default="True"/> <!-- if true pybullet gui will show -->
    <arg name="gui_options" default=""/>
    <arg name="rviz_bringup" default="False"/>
    <arg name="rviz_config" default="$(find pybullet_ros)/ros/config/rviz/pybullet_config.rviz"/>
    <arg name="start_paused" default="False"/> <!-- if true, will boot with paused physics -->
    <arg name="parallel_plugin_execution" default="True"/>
    <arg name="use_deformable_world" default="False"/> <!-- enable/disable soft body simulation -->

    <!-- pybullet ros wrapper node -->
    <node pkg="pybullet_ros" type="pybullet_ros_node" name="pybullet_ros" output="screen" required="true">
        <rosparam file="$(arg config_file)" command="load"/>
        <param name="plugin_import_prefix" value="$(arg plugin_import_prefix)"/>
        <param name="environment" value="$(arg environment)"/>
        <param name="pybullet_gui" value="$(arg pybullet_gui)"/>
        <param name="gui_options" value="$(arg gui_options)"/>
        <param name="start_paused" value="$(arg start_paused)"/>
        <param name="parallel_plugin_execution" value="$(arg parallel_plugin_execution)"/>
        <param name="use_deformable_world" value="$(arg use_deformable_world)"/>
    </node>

    <group if="$(arg rviz_bringup)">
        <node type="rviz" name="rviz" pkg="rviz" args="-d $(arg rviz_config)"/>
    </group>

</launch>
